cpp/src/transform/row_bit_count.cu

```diff
@@ -0,0 +1,508 @@
+      column_device_view const& offsets = col.child(lists_column_view::offsets_column_index);
+      auto const base_offset            = offsets.data<offset_type>()[col.offset()];
+      cur_span.row_start =
+        offsets.data<offset_type>()[cur_span.row_start + col.offset()] - base_offset;
+      cur_span.row_end = offsets.data<offset_type>()[cur_span.row_end + col.offset()] - base_offset;
+    }
+
+    last_branch_depth = info[idx].branch_depth_end;
+  }
+}
+
+}  // anonymous namespace
+
+/**
+ * @copydoc cudf::detail::row_bit_count
+ *
+ */
+std::unique_ptr<column> row_bit_count(table_view const& t,
+                                      rmm::cuda_stream_view stream,
+                                      rmm::mr::device_memory_resource* mr)
+{
+  // flatten the hierarchy and determine some information about it.
+  std::vector<cudf::column_view> cols;
+  std::vector<column_info> info;
+  hierarchy_info h_info;
+  flatten_hierarchy(t.begin(), t.end(), cols, info, h_info, stream);
+
+  // create output buffer and view
+  auto output = cudf::make_fixed_width_column(
+    data_type{type_id::INT32}, t.num_rows(), mask_state::UNALLOCATED, stream, mr);
+  mutable_column_view mcv = output->mutable_view();
+
+  // simple case.  if we have no complex types (lists, strings, etc), the per-row size is already
+  // trivially computed
+  if (h_info.complex_type_count <= 0) {
+    thrust::fill(rmm::exec_policy(stream),
+                 mcv.begin<size_type>(),
+                 mcv.end<size_type>(),
+                 h_info.simple_per_row_size);
+    return output;
+  }
+
+  // create a contiguous block of column_device_views
+  auto d_cols = contiguous_copy_column_device_views<column_device_view>(cols, stream);
+
+  // move stack info to the gpu
+  rmm::device_uvector<column_info> d_info(info.size(), stream);
+  CUDA_TRY(cudaMemcpyAsync(d_info.data(),
+                           info.data(),
+                           sizeof(column_info) * info.size(),
+                           cudaMemcpyHostToDevice,
+                           stream.value()));
+  CUDF_EXPECTS(info.size() == cols.size(), "Size/info mismatch");
+
+  // each thread needs to maintain a stack of row spans of size max_branch_depth. we will use
+  // shared memory to do this rather than allocating a potentially gigantic temporary buffer
+  // of memory of size (# input rows * sizeof(row_span) * max_branch_depth).
+  auto const shmem_per_thread = sizeof(row_span) * h_info.max_branch_depth;
+  int device_id;
+  CUDA_TRY(cudaGetDevice(&device_id));
+  int shmem_limit_per_block;
+  CUDA_TRY(
+    cudaDeviceGetAttribute(&shmem_limit_per_block, cudaDevAttrMaxSharedMemoryPerBlock, device_id));
+  constexpr int max_block_size = 256;
+  auto const block_size =
+    shmem_per_thread != 0
+      ? std::min(max_block_size, shmem_limit_per_block / static_cast<int>(shmem_per_thread))
+      : max_block_size;
+  auto const shared_mem_size = shmem_per_thread * block_size;
```

Comment:
This looks like a cool utility. Maybe consider adding it to `cudf/detail/utilities/cuda.cuh`.